### ISSUE TITLE
Channel has been shut down

### ISSUE THREAD

**Issue by timlukastlt** [pulumi/pulumi#16853]
### What happened?

If i execute pulumi preview or pulumi up i get the following error, but pulumi destroys works as aspected. 
For testing i created a fresh and new pulumi project with _pulumi new aws-javascript_ and their i also get the error without changing anything.

```bash
[user@workstation-al23 simple-pulumi]$ pulumi up
Please choose a stack, or create a new one: <create a new stack>
Please enter your desired stack name: dev 
Created stack 'dev'
Enter your passphrase to unlock config/secrets
    (set PULUMI_CONFIG_PASSPHRASE or PULUMI_CONFIG_PASSPHRASE_FILE to remember):  
Enter your passphrase to unlock config/secrets
Previewing update (dev):
     Type                 Name               Plan     Info
     pulumi:pulumi:Stack  simple-pulumi-dev           26 messages

Diagnostics:
  pulumi:pulumi:Stack (simple-pulumi-dev):
    failed to deliver log message.
    error: Error: Channel has been shut down
    original message: Running program '/home/emea1/user/repos/simple-pulumi' failed with an unhandled exception:
    Error: 1 CANCELLED: Call cancelled
        at callErrorFromStatus (/home/emea1/user/repos/simple-pulumi/node_modules/@grpc/grpc-js/build/src/call.js:31:19)
        at Object.onReceiveStatus (/home/emea1/user/repos/simple-pulumi/node_modules/@grpc/grpc-js/build/src/client.js:193:76)
        at Object.onReceiveStatus (/home/emea1/user/repos/simple-pulumi/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:360:141)
        at Object.onReceiveStatus (/home/emea1/user/repos/simple-pulumi/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:323:181)
        at /home/emea1/user/repos/simple-pulumi/node_modules/@grpc/grpc-js/build/src/resolving-call.js:129:78
        at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
    for call at
        at ServiceClientImpl.makeUnaryRequest (/home/emea1/user/repos/simple-pulumi/node_modules/@grpc/grpc-js/build/src/client.js:161:32)
        at ServiceClientImpl.supportsFeature (/home/emea1/user/repos/simple-pulumi/node_modules/@grpc/grpc-js/build/src/make-client.js:105:19)
        at result (/home/emea1/user/repos/simple-pulumi/node_modules/@pulumi/pulumi/runtime/settings.js:126:27)
        at new Promise (<anonymous>)
        at /home/emea1/user/repos/simple-pulumi/node_modules/@pulumi/pulumi/runtime/settings.js:125:30
        at Generator.next (<anonymous>)
        at /home/emea1/user/repos/simple-pulumi/node_modules/@pulumi/pulumi/runtime/settings.js:21:71
        at new Promise (<anonymous>)
        at __awaiter (/home/emea1/user/repos/simple-pulumi/node_modules/@pulumi/pulumi/runtime/settings.js:17:12)
        at monitorSupportsFeature (/home/emea1/user/repos/simple-pulumi/node_modules/@pulumi/pulumi/runtime/settings.js:122:12) {
      code: 1,
      details: 'Call cancelled',
      metadata: Metadata { internalRepr: Map(0) {}, options: {} }
    }
     message severity: error

Resources:

### Example
```

Standard template
```javascript
const pulumi = require("@pulumi/pulumi");
const aws = require("@pulumi/aws");
const awsx = require("@pulumi/awsx");

// Create an AWS resource (S3 Bucket)
const bucket = new aws.s3.Bucket("my-bucket");

// Export the name of the bucket
exports.bucketName = bucket.id;
```

### Output of `pulumi about`

CLI          
Version      3.122.0
Go Version   go1.22.4
Go Compiler  gc

Plugins
KIND      NAME    VERSION
resource  aws     6.47.0
resource  awsx    2.13.0
resource  docker  4.5.5
resource  docker  3.6.1
language  nodejs  unknown

Host     
OS       amazon
Version  2023.5.20240722
Arch     x86_64

This project is written in nodejs: executable='/home/emea1/user/.nvm/versions/node/v20.16.0/bin/node' version='v20.16.0'

Backend        
Name           workstation-al23.********.**********.**
URL            s3://************/test-delete-me-if-you-see-me
User           user
Organizations  
Token type     personal

Dependencies:
NAME            VERSION
@pulumi/pulumi  3.127.0
@pulumi/aws     6.47.0
@pulumi/awsx    2.13.0

Pulumi locates its logs in /tmp by default

node -v:
v20.16.0

npm -v:
10.8.1

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by timlukastlt** [comment]
My idea is that it has to do with Amazon Linux 2023, because with Debian it works.

**Comment by justinvp** [comment]
Sorry for the trouble and thanks for opening an issue. I haven't been able to reproduce this either. I'm not sure what was going on. I'm going to close this as no-repro, but feel free to open a new issue if you run into it again.

**Comment by t0yv0** [comment]
I have something similarly looking today in review stacks

```
Diagnostics:
  pulumi:pulumi:Stack (pulumi-service-review-stacks-t0yv0-dev):
    error: update failed

    (node:10800) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [process]. MaxListeners is 10. Use emitter.setMaxListeners() to increase limit
    (Use `node --trace-warnings ...` to show where the warning was created)
    failed to deliver log message.
    error: Error: Channel has been shut down
    original message: Error: 14 UNAVAILABLE: resource monitor shut down while waiting on output step's done channel
        at callErrorFromStatus (/Users/anton/code/pulumi-service/infrastructure/reviewStacks/node_modules/@grpc/grpc-js/src/call.ts:82:17)
        at Object.onReceiveStatus (/Users/anton/code/pulumi-service/infrastructure/reviewStacks/node_modules/@grpc/grpc-js/src/client.ts:360:55)
        at Object.onReceiveStatus (/Users/anton/code/pulumi-service/infrastructure/reviewStacks/node_modules/@grpc/grpc-js/src/client-interceptors.ts:458:34)
        at Object.onReceiveStatus (/Users/anton/code/pulumi-service/infrastructure/reviewStacks/node_modules/@grpc/grpc-js/src/client-interceptors.ts:419:48)
        at /Users/anton/code/pulumi-service/infrastructure/reviewStacks/node_modules/@grpc/grpc-js/src/resolving-call.ts:169:24
        at processTicksAndRejections (node:internal/process/task_queues:85:11)
    for call at
        at ServiceClientImpl.makeUnaryRequest (/Users/anton/code/pulumi-service/infrastructure/reviewStacks/node_modules/@grpc/grpc-js/src/client.ts:325:42)
        at ServiceClientImpl.registerResourceOutputs (/Users/anton/code/pulumi-service/infrastructure/reviewStacks/node_modules/@grpc/grpc-js/src/make-client.ts:189:15)        at /Users/anton/code/pulumi-service/infrastructure/reviewStacks/node_modules/@pulumi/runtime/resource.ts:1171:37
        at new Promise (<anonymous>)
        at /Users/anton/code/pulumi-service/infrastructure/reviewStacks/node_modules/@pulumi/runtime/resource.ts:1170:25
        at processTicksAndRejections (node:internal/process/task_queues:105:5)
     message severity: error

  docker-build:index:Image (service-amd64):
    error: failed to solve: process "/bin/sh -c GOOS=linux GOARCH=amd64     go build -v     -mod=vendor     -ldflags=\"-s -w\"     -o /go/bin/pulumi-insights-scanner     ./cmd/pulumi-insights-scanner" did not complete successfully: exit code: 1

    [Pulumi Copilot] Would you like help with these diagnostics?
    https://app.pulumi.com/pulumi/pulumi-service-review-stacks/t0yv0-dev/updates/20?explainFailure

```

**Comment by bkw** [comment]
Vote to reopen. This happens to us all the time now. The stack in question admittedly is way too large (3k+ resources), but it still seems like a bug, like disconnectSync not draining the event loop. I can supply more evidence and logs if you like.
